client/src/dataset/components/ScanSize.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from "react";
+import { Button, Form } from "semantic-ui-react";
+import { parseNumListWithPadding } from "../helpers";
+import ScanSizePart from "./ScanSizePart";
+
+interface ScanSizeProps {
+    value: string,
+    minScan: number,
+    maxScan: number,
+    setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void,
+}
+
+const ScanSize: React.FC<ScanSizeProps> = ({ value, minScan, maxScan, setFieldValue }) => {
+
+    const [scanSize, setScanSize] = React.useState(parseNumListWithPadding(value, minScan));
+
+    React.useEffect(() => {
+      const newScanSizeValue = parseNumListWithPadding(value, minScan);
+      setScanSize(newScanSizeValue);
+    }, [value, minScan]);
```

Comment:
You can also completely get rid of the `scanSize` state variable. Then you can just do something like `const scanSize = parseNumList...(value, minScan)`, which will get executed on each re-render, so every time `value` changes, `scanSize` will adapt. then below, you can remove all the `setScanSize` calls and only call `setFieldValue`. What do you think?

Reply:
> You can also completely get rid of the scanSize state variable.

That did occur to me, but I wasn't sure if the +/- functionality would work without having states.

> Then you can just do something like const scanSize = parseNumList...(value, minScan), which will get executed on each re-render, so every time value changes, scanSize will adapt.

Nice! So a state isn't needed at all, I've taken note of that, thanks!

Reply:
> I wasn't sure if the +/- functionality would work without having states.

Yes, this should work! As long as what you feed into `setFieldValue` round-trips together with your parse function, +/- should work (clicking "+" should result in a trailing comma)